--- poke-1.3.orig/poke/pk-repl.c
+++ poke-1.3/poke/pk-repl.c
@@ -329,18 +329,21 @@
   char *poke_history = NULL;
   /* Load the user's history file ~/.poke_history, if it exists
      in the HOME directory.  */
+  char *datadir = getenv ("XDG_DATA_HOME");
   char *homedir = getenv ("HOME");
 
-  if (homedir != NULL)
+  if (!poke_history && datadir != NULL)
     {
-      if (asprintf (&poke_history, "%s/.poke_history", homedir) != -1)
-        {
-          if (access (poke_history, R_OK) == 0)
-            read_history (poke_history);
-        }
-      else
+      if (asprintf (&poke_history, "%s/poke_history", datadir) == -1)
         poke_history = NULL;
     }
+  if (!poke_history && homedir != NULL)
+    {
+      if (asprintf (&poke_history, "%s/.local/share/poke_history", homedir) == -1)
+        poke_history = NULL;
+    }
+  if (poke_history && access (poke_history, R_OK) == 0)
+    read_history (poke_history);
 #endif
   rl_getc_function = poke_getc;
   rl_completer_quote_characters = "\"";
