+++ ./plugins/trayicon.py
@@ -109,11 +109,7 @@
 
         self.window = self.shell.get_window()
         self.delete_signal_id = GObject.signal_lookup("delete_event", Gtk.Window)
-        GObject.signal_handlers_block_matched (self.window,
-                                               GObject.SignalMatchType.ID | GObject.SignalMatchType.DATA,
-                                               self.delete_signal_id, 0, None, None, None)
         self.window.connect("delete_event", self.trayicon_minimize_on_close)
-        self.window.connect("window-state-event", self.window_state_event_cb)
 
         # show the window if it is hidden when starting liferea
         self.window.deiconify()
@@ -125,12 +121,6 @@
         self.feedlist_new_items_cb_id = sigid
         self.feedlist = feedlist
 
-    def window_state_event_cb(self, widget, event):
-        "Hide window when minimize"
-        if event.changed_mask & event.new_window_state & Gdk.WindowState.ICONIFIED:
-            self.window.deiconify()
-            self.window.hide()
-
     def trayicon_click(self, widget, data = None):
         self.window.deiconify()
         self.window.show()
@@ -189,10 +179,6 @@
     def do_deactivate(self):
         self.staticon.set_visible(False)
         self.window.disconnect_by_func(self.trayicon_minimize_on_close)
-        GObject.signal_handlers_unblock_matched (self.window,
-                                                 GObject.SignalMatchType.ID | GObject.SignalMatchType.DATA,
-                                                 self.delete_signal_id, 0, None,None,None)
-        self.window.disconnect_by_func(self.window_state_event_cb)
 
         self.feedlist.disconnect(self.feedlist_new_items_cb_id)
 
