--- wget-1.21.1.orig/src/main.c
+++ wget-1.21.1/src/main.c
@@ -176,9 +176,16 @@ get_hsts_database (void)
   if (opt.hsts_file)
     return xstrdup (opt.hsts_file);
 
+  char *cache_dir = getenv("XDG_CACHE_DIR");
+  if (cache_dir)
+    {
+      char *dir = aprintf("%s/wget-hsts", cache_dir);
+      return dir;
+    }
+
   if (opt.homedir)
     {
-      char *dir = ajoin_dir_file(opt.homedir, ".wget-hsts");
+      char *dir = aprintf("%s/.cache/wget-hsts", opt.homedir);
       return dir;
     }
 
