Disable smooth scrolling

--- zapzap-6.0-beta-8.orig/zapzap/controllers/WebView.py
+++ zapzap-6.0-beta-8/zapzap/controllers/WebView.py
@@ -63,7 +63,7 @@
             lambda notification: NotificationManager.show(self, notification)
         )
         self.profile.settings().setAttribute(
-            QWebEngineSettings.WebAttribute.ScrollAnimatorEnabled, True)
+            QWebEngineSettings.WebAttribute.ScrollAnimatorEnabled, False)
 
         self.configure_spellcheck()
 
